tests/impl.cpp

```diff
@@ -9228,14 +9228,15 @@ result_t test_mm_test_mix_ones_zeros(const SSE2NEONTestImpl &impl,
     __m128i a = load_m128i(_a);
     __m128i mask = load_m128i(_mask);
 
-    int32_t d0 = !((_a[0]) & _mask[0]) & !((!_a[0]) & _mask[0]);
-    int32_t d1 = !((_a[1]) & _mask[1]) & !((!_a[1]) & _mask[1]);
-    int32_t d2 = !((_a[2]) & _mask[2]) & !((!_a[2]) & _mask[2]);
-    int32_t d3 = !((_a[3]) & _mask[3]) & !((!_a[3]) & _mask[3]);
-    int32_t result = ((d0 & d1 & d2 & d3) == 0) ? 1 : 0;
+    int32_t has_ones = 0;
+    int32_t has_zeros = 0;
+    for (int i = 0; i < 4; i++) {
+        has_ones |= _a[i] & _mask[i];
+        has_zeros |= ~_a[i] & _mask[i];
+    }
+    int32_t result = ((has_ones != 0) & (has_zeros != 0));
```

Comment:
If possible maybe we can add some edge cases here to test the behavior?

Reply:
I'm not sure where the values are coming from, I'll leave this to you.

Reply:
> If possible maybe we can add some edge cases here to test the behavior?

@howjmay, Can you show relevant code accordingly?

Reply:
I think if there is a test case all the bits of `mask` is zero would be good. It is a case that force the answer to be zero